pyblish/util.py

```diff
@@ -123,11 +116,13 @@ def _convenience_iter(context=None, plugins=None, targets=None, order=None):
     # dynamically determined at run-time by contents of
     # the context and families of contained instances;
     # each of which may differ between task.
-    task_count = len(list(logic.Iterator(plugins, context)))
+    task_count = len(list(logic.Iterator(plugins, context, targets=targets)))
 
     # First pass, collection
     tasks_processed_count = 1
-    for Plugin, instance in logic.Iterator(collectors, context):
+    for Plugin, instance in logic.Iterator(
+        collectors, context, targets=targets
+    ):
```

Comment:
Stylistic note, please make this..

```py
    for Plugin, instance in logic.Iterator(collectors,
                                           context,
                                           targets=targets):
        ...
```

Alternatively..

```py
it = logic.Iterator(collectors, context, targets=targets)
for Plugin, instance in it:
   ...
```

Reply:
Updated